src/main/java/org/broadinstitute/barclay/help/DefaultDocWorkUnitHandler.java

```diff
@@ -191,19 +191,20 @@ public void processWorkUnit(
         // Ex: We may want to document an input or output file format by adding @DocumentedFeature
         // to the format's reader/writer class (i.e. TableReader), and then reference that feature
         // in the extraDocs attribute in a CommandLineProgram that reads/writes that format.
-        if (workUnit.getCommandLineProperties() != null) {
-            try {
-                final Object argumentContainer = workUnit.getClazz().newInstance();
-                if (argumentContainer instanceof CommandLinePluginProvider ) {
-                    pluginDescriptors = ((CommandLinePluginProvider) argumentContainer).getPluginDescriptors();
-                    clp = new CommandLineArgumentParser(
-                            argumentContainer, pluginDescriptors, Collections.emptySet()
-                    );
-                } else {
-                    clp = new CommandLineArgumentParser(argumentContainer);
-                }
-            } catch (IllegalAccessException | InstantiationException e) {
-                throw new RuntimeException(e);
+        try {
+            final Object argumentContainer = workUnit.getClazz().newInstance();
+            if (argumentContainer instanceof CommandLinePluginProvider) {
+                pluginDescriptors = ((CommandLinePluginProvider) argumentContainer).getPluginDescriptors();
+                clp = new CommandLineArgumentParser(
+                        argumentContainer, pluginDescriptors, Collections.emptySet()
+                );
+            } else {
+                clp = new CommandLineArgumentParser(argumentContainer);
+            }
+        } catch (IllegalAccessException | InstantiationException e) {
+            // only throw if the command line properties is set - CLP always require a non-arg constructor
```

Comment:
Change this comment to say "DocumentedFeature does not assume a no-arg constructor unless it is also a CommandLineProgram".

Reply:
Done.
(Because there is no concept of `CommandLineProgram` yet in Barclay, but only the annotation, I changed the ref to `CommandLineProgramProperties` instead)